### ISSUE TITLE
Feature: Single name prompt for sign in

### ISSUE THREAD

**Issue by sergeant-savage** [sergeant-savage/chatter#11]
I think that using `prompt()` method defined in JavaScript, we can restrict the setting of names, and also thus remove the necessity to impersonate other people (except by actual coordination).

**Comment by sergeant-savage** [comment]
I'm honestly not sure this is worth pursuing anymore. Thoughts?

**Comment by Glacier150** [comment]
I don't think we need an account system, but preventing duplicate Usernames would still be nice, we could maybe find a way to get a real time list of usernames to do something like this.

**Comment by sergeant-savage** [comment]
@Glacier150 Good idea. I was thinking of embedding some state by creating a struct called `ConnectionFactory` that implemented the `Factory` trait and putting new connections inside a `Vec`. Then we could implement the `client_connected` method, and have the handler broadcast a room update before being returned as part of the signature.
